File: Domain/ListingAggregate/Objects/Areas.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;

namespace Domain.ListingAggregate.Objects
{
    public class Areas
    {
        public Guid Id { get; set; }
        public Area External { get; set; }
        public Area Internal { get; set; }
        public Guid ListingId { get; set; }
        public Listing Listing { get; set; }
    }
}

File: client-app/src/app/model/ListingAggregate/Objects/Areas.ts
import { Listing } from "../Listing";

export interface Areas {
    id: string;
    external: Area;
    internal: Area;
    listingId: string;
    listing: Listing;
}

export interface Area {
    value: number;
    units: UnitOfArea;
}

export enum UnitOfArea {
    SqFeet,
    SqYard,
    SqMetres,
    Acres,
    Hectares
}